test/core/event_engine/posix/posix_engine_listener_utils_test.cc

```diff
@@ -0,0 +1,162 @@
+
+#include <grpc/support/log.h>
+
+#include "src/core/lib/event_engine/channel_args_endpoint_config.h"
+#include "src/core/lib/event_engine/posix_engine/posix_engine_listener_utils.h"
+#include "src/core/lib/event_engine/posix_engine/tcp_socket_utils.h"
+#include "test/core/util/port.h"
+
+namespace grpc_event_engine {
+namespace posix_engine {
+
+namespace {
+
+using ::grpc_event_engine::experimental::ChannelArgsEndpointConfig;
+
+class TestListenerSocketsContainer : public ListenerSocketsContainer {
+ public:
+  void Append(ListenerSocket socket) override { sockets_.push_back(socket); }
+
+  absl::StatusOr<ListenerSocket> Find(
+      const grpc_event_engine::experimental::EventEngine::ResolvedAddress& addr)
+      override {
+    for (auto socket = sockets_.begin(); socket != sockets_.end(); ++socket) {
+      if (socket->addr.size() == addr.size() &&
+          memcmp(socket->addr.address(), addr.address(), addr.size()) == 0) {
+        return *socket;
+      }
+    }
+    return absl::NotFoundError("Socket not found!");
+  }
+
+  int Size() { return static_cast<int>(sockets_.size()); }
+
+  std::list<ListenerSocket>::const_iterator begin() { return sockets_.begin(); }
+  std::list<ListenerSocket>::const_iterator end() { return sockets_.end(); }
+
+ private:
+  std::list<ListenerSocket> sockets_;
+};
+
+}  // namespace
+
+TEST(PosixEngineListenerUtils, ListenerContainerAddWildcardAddressesTest) {
+  TestListenerSocketsContainer listener_sockets;
+  int port = grpc_pick_unused_port_or_die();
+  ChannelArgsEndpointConfig config;
+  auto result = ListenerContainerAddWildcardAddresses(
+      listener_sockets, TcpOptionsFromEndpointConfig(config), port);
+  EXPECT_TRUE(result.ok());
+  EXPECT_GT(*result, 0);
+  port = *result;
+  EXPECT_GE(listener_sockets.Size(), 1);
+  EXPECT_LE(listener_sockets.Size(), 2);
+  for (auto socket = listener_sockets.begin(); socket != listener_sockets.end();
+       ++socket) {
+    ASSERT_TRUE((*socket).addr.address()->sa_family == AF_INET6 ||
+                (*socket).addr.address()->sa_family == AF_INET);
+    if ((*socket).addr.address()->sa_family == AF_INET6) {
+      EXPECT_EQ(SockaddrToString(&(*socket).addr, true).value(),
+                absl::StrCat("[::]:", std::to_string(port)));
+    } else if ((*socket).addr.address()->sa_family == AF_INET) {
+      EXPECT_EQ(SockaddrToString(&(*socket).addr, true).value(),
+                absl::StrCat("0.0.0.0:", std::to_string(port)));
+    }
+    close(socket->sock.Fd());
+  }
+}
+
+#ifdef GRPC_HAVE_IFADDRS
+TEST(PosixEngineListenerUtils, ListenerContainerAddAllLocalAddressesTest) {
+  TestListenerSocketsContainer listener_sockets;
+  int port = grpc_pick_unused_port_or_die();
+  ChannelArgsEndpointConfig config;
+  struct ifaddrs* ifa = nullptr;
+  struct ifaddrs* ifa_it;
+  if (getifaddrs(&ifa) != 0 || ifa == nullptr) {
+    // No ifaddresses available.
+    gpr_log(GPR_INFO,
+            "Skipping ListenerAddAllLocalAddressesTest because the machine "
```

Comment:
No fix needed, I'm curious: which environments does this apply to?

Reply:
If the test runs on a machine with no interfaces I think this condition could get triggered. Looks like it could also get triggered if there is no memory